streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala

```diff
@@ -576,7 +576,8 @@ abstract class DStream[T: ClassTag] (
    * of this DStream.
    */
   def reduce(reduceFunc: (T, T) => T): DStream[T] = ssc.withScope {
-    this.map(x => (null, x)).reduceByKey(reduceFunc, 1).map(_._2)
+    val cleanedReduceFunc = context.sparkContext.clean(reduceFunc)
+    this.map(x => (null, x)).reduceByKey(cleanedReduceFunc, 1).map(_._2)
```

Comment:
This one's actually not necessary because it will be cleaned by `reduceByKey` later